src/roku/roku-api.ts

```diff
@@ -0,0 +1,148 @@
+    });
+
+    const outputSigningPath = path.join(outputPath)
+    const outputPackagePath = path.join(outputSigningPath, `${packageName}${PACKAGE_EXTENSION}`)
+    const outputCredentialsPath = path.join(outputSigningPath, CREDENTIALS_FILENAME)
+
+    if (!fs.existsSync(outputSigningPath)) {
+        fs.mkdirSync(outputSigningPath, { recursive: true })
+    }
+    fs.copyFileSync(packagePath, outputPackagePath)
+    fs.writeFileSync(outputCredentialsPath, JSON.stringify(signingProperties))
+
+    return outputPath
+}
+
+export async function executeDeviceRekey(signingPath: string, deviceProperties?: DeviceProperties) {
+    
+    const signingProperties = parseSigningProperties(signingPath)
+
+    await rekeyDevice({
+        ...getDeviceProperties(deviceProperties?.device, deviceProperties?.password, deviceProperties?.username),
+        signingPassword: signingProperties.credentials.password,
+        rekeySignedPackage: path.resolve(signingProperties.packageFilePath),
+        devId: signingProperties.credentials.dev_id
+    })
+}
+
+function parseSigningProperties(signingPropertiesPath: string) {
+    var files = fs.readdirSync(signingPropertiesPath);
+    const packageFilename = files.find((filePath) => path.extname(filePath) == PACKAGE_EXTENSION);
+    if (!packageFilename) {
+        throw Error(`Could not find package at path: ${signingPropertiesPath}`)
+    }
+    const packageFilePath = path.join(signingPropertiesPath, packageFilename)
+
+    const credentialsFilePath = path.join(signingPropertiesPath, CREDENTIALS_FILENAME)
+    if (!path.resolve(credentialsFilePath)) {
+        throw Error(`Could not find credentials.json at path: ${credentialsFilePath}`)
+    }
+
+    const credentialsData = fs.readFileSync(credentialsFilePath, 'utf-8')
+    try {
+        const credentials = JSON.parse(credentialsData)
+        if (credentials.dev_id && credentials.password) {
+            return { credentials, packageFilePath }
+        } else {
+
+            throw Error(`Missing required keys on credentials.json: dev_id, password`)
+        }
+    } catch (error) {
+        throw Error(`Could not parse credentials file: ${credentialsFilePath}`)
+    }
+}
+
+function getResourceAt(resourcePath: string) {
+    const packagePath = '${__dirname}'
+    const resourcesFolderPath = path.join(packagePath, RESOURCE_FOLDER_PATH)
+    return path.resolve(path.join(resourcesFolderPath, resourcePath))
+}
+
+function getDeviceProperties(device: string | undefined = undefined, password: string | undefined = undefined, username: string | undefined = undefined) {
+    const env = process.env
+    const finalHost = device ? device : env.ROKU_DEVICE_ADDRESS
+    const finalUsername = username ? username : env.ROKU_DEVICE_USERNAME
+    const finalPassword = password ? password : env.ROKU_DEVICE_PASSWORD
+
+    if (finalHost && finalUsername && finalPassword) {
+        return { host:finalHost, username:finalUsername, password:finalPassword }
```

Comment:
I would recommend change the format to 
`{ host: finalHost, username: finalUsername, password: finalPassword }`